opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -2583,6 +2583,17 @@ namespace Opm {
         this->prod_index_calc_[well_index].reInit(well);
     }
 
+    template<typename TypeTag>
+    void
+    BlackoilWellModel<TypeTag>::
+    updateEclWell(int timeStepIdx, const std::string& wname) {
+        auto well_iter = std::find_if( this->wells_ecl_.begin(), this->wells_ecl_.end(), [wname] (const auto& well) -> bool { return well.name() == wname;});
+        if (well_iter == this->wells_ecl_.end())
+            throw std::logic_error("Could not find well: " + wname);
+
```

Comment:
Same code is in this file at least another time. Maybe use add a functions for reuse?